Source/DafnyDriver/Commands/RunCommand.cs

```diff
@@ -30,6 +30,7 @@ class RunCommand : ICommandSpec {
   public IEnumerable<Option> Options =>
     new Option[] {
       Inputs,
+      CommonOptionBag.Output
```

Comment:
Are you using this option? I think the concept of `output` for `dafny run` is different than for existing commands where it occurs, since run doesn't have producing files as a purpose. It just has intermediate files it may produce to run the program.

Could the ideal be that we have a build folder for intermediate artifacts, and a separate output folder for commands that produce output files?

Reply:
As you can probably tell, I need it to hit the bug this PR fixes. We could rewrite the compiler tester to manually run the generated sources, but I don't think this duplication of functionality is worth keeping the options list smaller.

If you go down the route you've sketched, I suggest generating a unique folder name and deleting it after finishing running.

Reply:
> As you can probably tell, I need it to hit the bug this PR fixes. We could rewrite the compiler tester to manually run the generated sources, but I don't think this duplication of functionality is worth keeping the options list smaller.

Could you add this option using a new option that's hidden? That way, you can use it in the test but it's not exposed in the public API. Also, maybe you can already make it be named `--build` so it's a little like what I sketched.

> If you go down the route you've sketched, I suggest generating a unique folder name and deleting it after finishing running. 

If we go down that route? :D

I think a unique folder that's deleted after running would be good yes, as a behavior for when you don't specify an explicit build directory.